modules/data.atmosphere/R/met2cf.ERA5.R

```diff
@@ -0,0 +1,171 @@
+#' met2cf.ERA5
+#'
+#' @param lat latitude
+#' @param long longitude
+#' @param years years to be extracted
+#' @param overwrite Logical flag definign if the files need to be overwriten in case they were downloaded before.
+#' @param verbose Logical flag defining if ouput of function be extra verbose.
+#' @param sitename The name of the site used for making the identifier.
+#' @param data.folder Path to the directory where ERA5 nc files are located.
+#' @param outfolder Path to directory where nc files need to be saved.
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' files.era5 <- met2cf.ERA5(years=2018,
+#'                   data.folder=".",
+#'                   outfolder = "./metfiles")
+#'
+#' }
+met2cf.ERA5 <- function(lat = 40,
+                        long = -120,
+                        years = c(2015:2017),
+                        overwrite = FALSE,
+                        verbose = T,
+                        sitename,
+                        data.folder,
+                        outfolder) {
+  
+  # Extracting the raw data - The output would be a list of xts objects for each ensemble
+  out <- ERA5_extract_ENS(lat = lat,
+                          long = long,
+                          years = years, 
+                          data.folder=data.folder)
+  #start and end date
+  start_date <- min(index(out[[1]]))
+  end_date <- max(index(out[[1]]))
+  
+  # adding RH and converting rain
+  out.new <- names(out) %>%
+    purrr::map(function(ensi) {
+ 
+      ens <- out[[ensi]]
+      # Solar radation conversions 
+      #https://confluence.ecmwf.int/pages/viewpage.action?pageId=104241513
+      #For ERA5 daily ensemble data, the accumulation period is 3 hours. Hence to convert to W/m2:
+      ens[, "ssrd"] <- ens[, "ssrd"]/ (3 * 3600)
+      ens[, "strd"] <- ens[, "strd"]/ (3 * 3600)
+      #precipitation it's originaly in meters. Meters times the density will give us the kg/m2
+      ens[, "tp"] <- ens[, "tp"]*1000/3 # divided by 3 because we have 3 hours data
+      ens[, "tp"] <- udunits2::ud.convert(ens[, "tp"], "kg m-2 hr-1", "kg m-2 6 s-1")  #There are 21600 seconds in 6 hours
+      #RH
+      #Adopted from weathermetrics/R/moisture_conversions.R
+      t <- udunits2::ud.convert(ens[, "t2m"] %>% as.numeric(), "K", "degC")
+      dp <- udunits2::ud.convert(ens[, "d2m"] %>% as.numeric(), "K", "degC")
+      beta <- (112 - (0.1 * t) + dp) / (112 + (0.9 * t))
```

Comment:
Please change `dp` to `dewpoint` to make the intent of this code clearer.

Reply:
Also, I'm a bit wary of cavalier use of `as.numeric` in places like this. If the `ens` object is not numeric, that almost certainly means there was a problem upstream, and the code should fail here rather than performing R's often surprising and counterintuitive type conversion (e.g. `as.numeric(factor(c("4", "5", "6")) == c(1, 2, 3)`).